pkg/log/logger.go

```diff
@@ -0,0 +1,168 @@
+package log
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log/slog"
+	"os"
+	"regexp"
+
+	"github.com/lmittmann/tint"
+)
+
+const (
+	levelTrace = slog.Level(-8)
+	levelDebug = slog.Level(-4)
+	levelInfo  = slog.Level(0)
+	levelWarn  = slog.Level(4)
+	levelError = slog.Level(8)
+	levelFatal = slog.Level(12)
+)
+
+const (
+	red     = "\033[31m"
+	green   = "\033[32m"
+	yellow  = "\033[33m"
+	blue    = "\033[34m"
+	magenta = "\033[35m"
+	cyan    = "\033[36m"
+	gray    = "\033[37m"
+	white   = "\033[97m"
+)
+
+type customLevel struct {
+	label string
+	color string
+}
+
+var customLevels = map[slog.Leveler]customLevel{
+	levelDebug: {
+		label: "DEBUG",
+		color: green,
+	},
+	levelInfo: {
+		label: "INFO",
+		color: blue,
+	},
+	levelWarn: {
+		label: "WARN",
+		color: yellow,
+	},
+	levelError: {
+		label: "ERROR",
+		color: red,
+	},
+	levelTrace: {
+		label: "TRACE",
+		color: magenta,
+	},
+	levelFatal: {
+		label: "FATAL",
+		color: blue,
+	},
+}
+
+type fileWriter struct {
+	w io.Writer
+}
+
+func (e fileWriter) Write(p []byte) (int, error) {
+
+	re := regexp.MustCompile(`\033[[0-9;]*m`)
+
+	// Trim colors from output
+	newStr := re.ReplaceAllString(string(p), "")
+
+	data := []byte(newStr)
+
+	n, err := e.w.Write(data)
+
+	if err != nil {
+		return n, err
+	}
+	if n != len(data) {
+		return n, io.ErrShortWrite
+	}
+
+	return len(p), nil
+}
+
+type LogConfig struct {
+	LogLevel    int
+	LogFileName string
+}
+
+var ctx context.Context
+var logger *slog.Logger
```

Comment:
I would drop this global. Otherwise it's not possible to create different loggers.